### ISSUE TITLE
fix crash in DropletActivity

### ISSUE THREAD

**Issue by championswimmer** [coding-blocks/DigitalOceanApp#93]
Logs

```
java.lang.NullPointerException: 
 
  at in.tosc.digitaloceanapp.activities.DropletActivity$4.onResponse (DropletActivity.java)
 
  at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run (ExecutorCallAdapterFactory.java)
 
  at android.os.Handler.handleCallback (Handler.java:751)
 
  at android.os.Handler.dispatchMessage (Handler.java:95)
 
  at android.os.Looper.loop (Looper.java:154)
 
  at android.app.ActivityThread.main (ActivityThread.java:6688)
 
  at java.lang.reflect.Method.invoke (Method.java)
 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1468)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1358)
```

**Comment by championswimmer** [comment]
Second log

```
java.lang.NullPointerException: 
 
  at in.tosc.digitaloceanapp.activities.DropletActivity$3.onResponse (DropletActivity.java)
 
  at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run (ExecutorCallAdapterFactory.java)
 
  at android.os.Handler.handleCallback (Handler.java:751)
 
  at android.os.Handler.dispatchMessage (Handler.java:95)
 
  at android.os.Looper.loop (Looper.java:154)
 
  at android.app.ActivityThread.main (ActivityThread.java:6688)
 
  at java.lang.reflect.Method.invoke (Method.java)
 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1468)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1358)
```

**Comment by championswimmer** [comment]
These are two different errors, both due to most probably request finishing after user navigating away. 

Both need to be fixed

**Comment by aggarwalpulkit596** [comment]
On it

**Comment by aggarwalpulkit596** [comment]
@championswimmer bhaiya I'm not able to replicate the error can you the steps to replicate the errors

**Comment by mohityv** [comment]
@championswimmer please review #95 pr as well

**Comment by aggarwalpulkit596** [comment]
![screen shot 2018-06-09 at 2 09 00 pm](https://user-images.githubusercontent.com/29139786/41190490-4b881212-6bfd-11e8-820a-88bbb59c60bc.png)
Some issue running travis test @championswimmer please check

**Comment by aggarwalpulkit596** [comment]
https://github.com/coding-blocks/DigitalOceanApp/issues/96 Travis has to be fixed first

**Comment by mohityv** [comment]
@championswimmer Travis errors solved in pr #95
